wgsl/index.bs

```diff
@@ -2709,7 +2709,12 @@ Variables in the [=storage classes/storage=] storage class and variables with a
 [storage texture](#texture-storage) type must have an [=access=] attribute
 applied to the store type.
 
+The <dfn noexport>lifetime</dfn> of a variable is determined by its scope.
```

Comment:
Lifetime is a dynamic context, but some of this reads like it's a static context.
It should also take into account function calls.  How about this:

> The lifetime of a variable is the period during shader execution for which the variable exists.
The lifetime of a module-scope variable is the entire execution of the shader stage.

> For a function-scope variable, each invocation has its own independent version of the variable.
> The lifetime of a function-scope variable is determined by its scope:
> * It begins when control enters the variable's scope.
> * It includes the entire execution of any function called from within the variable's scope.
> * It ends when control leaves the variable's scope, other than calling a function from within the variable's scope.